File: app/src/main/java/com/karthik/myweather/data/entities/Weather.java
package com.karthik.myweather.data.entities;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "weather")
public class Weather {
    public int locationId;
    @NonNull
    @PrimaryKey
    public String date;
    public double minTemperature;
    public double maxTemperature;
    public double windSpeed;
    public String weatherCode;
    public String weatherDescription;
}

File: app/src/main/java/com/karthik/myweather/data/entities/Weather.kt
package com.karthik.myweather.data.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "weather")
data class Weather (
        var locationId: Int = 0,
        @PrimaryKey
        var date: String,
        var minTemperature: Double = 0.0,
        var maxTemperature: Double = 0.0,
        var windSpeed: Double = 0.0,
        var weatherCode: String? = null,
        var weatherDescription: String? = null
)